Continued execution of the delete operation will cause the fe process to exit

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
execute delete operation on a non-master fe continuously.

### Expected behavior (Required)
System runs normally.

### Real behavior (Required)
Fe crashed with exception:
```java
2022-02-22 20:26:47,135 ERROR (starrocks-mysql-nio-pool-0|147) [EditLog.logEdit():864] Fatal Error : write stream Exception
com.sleepycat.je.rep.DatabasePreemptedException: (JE 7.3.7) (JE 7.3.7) Database 76036868 has been forcibly closed in order to apply a replicated remove operation.  This Database and all associated Cursors must be closed.  All associated Transactions must be aborted.
        at com.sleepycat.je.rep.DatabasePreemptedException.wrapSelf(DatabasePreemptedException.java:113) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.Database.checkOpen(Database.java:2274) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.Database.getDatabaseName(Database.java:2046) ~[je-7.3.7.jar:7.3.7]
        at com.starrocks.journal.bdbje.BDBJEJournal.write(BDBJEJournal.java:195) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.logEdit(EditLog.java:860) ~[starrocks-fe.jar:?]
        at com.starrocks.persist.EditLog.logSaveNextId(EditLog.java:908) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.CatalogIdGenerator.getNextId(CatalogIdGenerator.java:54) ~[starrocks-fe.jar:?]
        at com.starrocks.catalog.Catalog.getNextId(Catalog.java:5111) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.DeleteStmt.<init>(DeleteStmt.java:54) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.CUP$SqlParser$actions.case473(SqlParser.java:26231) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.CUP$SqlParser$actions.CUP$SqlParser$do_action(SqlParser.java:8951) ~[starrocks-fe.jar:?]
        at com.starrocks.analysis.SqlParser.do_action(SqlParser.java:1908) ~[starrocks-fe.jar:?]
        at java_cup.runtime.lr_parser.parse(lr_parser.java:584) ~[jflex-1.4.3.jar:?]
        at com.starrocks.common.util.SqlParserUtils.getMultiStmts(SqlParserUtils.java:63) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.analyze(ConnectProcessor.java:325) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:253) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.dispatch(ConnectProcessor.java:415) ~[starrocks-fe.jar:?]
        at com.starrocks.qe.ConnectProcessor.processOnce(ConnectProcessor.java:651) ~[starrocks-fe.jar:?]
        at com.starrocks.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:55) ~[starrocks-fe.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_202]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_202]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_202]
Caused by: com.sleepycat.je.rep.DatabasePreemptedException: (JE 7.3.7) Database 76036868 has been forcibly closed in order to apply a replicated remove operation.  This Database and all associated Cursors must be closed.  All associated Transactions must be aborted.
        at com.sleepycat.je.rep.impl.RepImpl.createDatabasePreemptedException(RepImpl.java:2008) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.RepImpl.createDatabasePreemptedException(RepImpl.java:143) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.Database.setPreempted(Database.java:469) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.DbInternal.setPreempted(DbInternal.java:58) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.dbi.DbTree.lockNameLN(DbTree.java:972) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.dbi.DbTree.doRemoveDb(DbTree.java:1172) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.dbi.DbTree.removeReplicaDb(DbTree.java:1239) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replay.applyNameLN(Replay.java:872) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replay.replayEntry(Replay.java:598) ~[je-7.3.7.jar:7.3.7]
        at com.sleepycat.je.rep.impl.node.Replica$ReplayThread.run(Replica.java:1213) ~[je-7.3.7.jar:7.3.7]
```

### StarRocks version (Required)
 - main

--- a/fe/fe-core/src/main/java/com/starrocks/analysis/DeleteStmt.java
+++ b/fe/fe-core/src/main/java/com/starrocks/analysis/DeleteStmt.java
@@ -44,18 +44,21 @@ public class DeleteStmt extends DdlStmt {
     private final List<Predicate> deleteConditions;
     // Each deleteStmt corresponds to a DeleteJob.
     // The JobID is generated here for easy correlation when cancel Delete
-    private final long jobId;
+    private long jobId = -1;
 
     public DeleteStmt(TableName tableName, PartitionNames partitionNames, Expr wherePredicate) {
         this.tbl = tableName;
         this.partitionNames = partitionNames;
         this.wherePredicate = wherePredicate;
         this.deleteConditions = Lists.newLinkedList();
-        this.jobId = Catalog.getCurrentCatalog().getNextId();
     }
 
     public long getJobId() {
         return jobId;
+    }
+
+    public void setJobId(long jobId) {
+        this.jobId = jobId;
     }
 
     public String getTableName() {

--- a/fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java
+++ b/fe/fe-core/src/main/java/com/starrocks/load/DeleteHandler.java
@@ -201,7 +201,8 @@ public class DeleteHandler implements Writable {
 
                 // generate label
                 String label = "delete_" + UUID.randomUUID();
-                long jobId = stmt.getJobId();
+                long jobId = Catalog.getCurrentCatalog().getNextId();
+                stmt.setJobId(jobId);
                 // begin txn here and generate txn id
                 transactionId = Catalog.getCurrentGlobalTransactionMgr().beginTransaction(db.getId(),
                         Lists.newArrayList(table.getId()), label, null,

--- a/fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java
+++ b/fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java
@@ -637,7 +637,10 @@ public class StmtExecutor {
     public void cancel() {
         if (parsedStmt instanceof DeleteStmt) {
             DeleteStmt deleteStmt = (DeleteStmt) parsedStmt;
-            Catalog.getCurrentCatalog().getDeleteHandler().killJob(deleteStmt.getJobId());
+            long jobId = deleteStmt.getJobId();
+            if (jobId != -1) {
+                Catalog.getCurrentCatalog().getDeleteHandler().killJob(jobId);
+            }
         } else {
             Coordinator coordRef = coord;
             if (coordRef != null) {

